src/tests/failureMessageTests/determinism_test.rs

```diff
@@ -0,0 +1,38 @@
+#[cfg(test)]
+
+mod test {
+    use crate::System::refine::{RefinementFailure, RefinementResult};
+    use crate::{
+        tests::refinement::Helper::json_run_query, System::executable_query::QueryResult,
+        System::local_consistency::DeterminismResult, TransitionSystems::LocationID,
+    };
+
+    static PATH: &str = "samples/json/Determinism";
+
+    #[test]
+    fn determinism_failure_test() {
+        let temp = json_run_query(PATH, "determinism: NonDeterminismCom");
+        assert!(
+            if let QueryResult::Determinism(DeterminismResult::Failure(LocationID::Simple(_))) =
+                temp
+            {
+                true
+            } else {
+                false
+            }
+        );
+    }
+
+    #[test]
+    fn determinism_failure_in_refinemnet_test() {
+        let temp = json_run_query(PATH, "refinement: NonDeterminismCom <= Component2");
+        assert!(if let QueryResult::Refinement(RefinementResult::Failure(
+            RefinementFailure::DeterminismFailure(Some(_)),
+        )) = temp
+        {
+            true
+        } else {
+            false
+        });
```

Comment:
Again this formatting looks very very strange

Reply:
I think you can use the macro `match!`

Reply:
Lækkert råd - Der er lavet commit hvor matches! benyttes istedet

Reply:
Ser sku godt ud